align/schema/checker.py

```diff
@@ -136,13 +136,20 @@ def __init__(self):
         self._bbox_cache = {}
         self._bbox_subcircuit = {}
         self._solver = z3.Solver()
+        self.previously_unsat = False
 
     def append(self, formula, identifier=None):
         self._solver.add(formula)
         r = self._solver.check()
         if r == z3.unsat:
             z3.set_option(max_depth=10000, max_args=100, max_lines=10000)
-            raise CheckerError(f'No solution exists for {formula} in conjunction with {self._solver}')
+            raise_on_unsat = True
+            if raise_on_unsat:
+                raise CheckerError(f'No solution exists for {formula} in conjunction with {self._solver}')
+            else:
+                if not self.previously_unsat:
```

Comment:
Does else statement ever executed?

Reply:
I was experimenting with not having a fatal error when there is a checker violation. I wanted to talk to you or Parijat about that. This was an attempt. (change raise_on_unsat to False and you get the other behavior.) I'll leave this in until we can have a discussion about it.